research-software-directory/RSD-as-a-service
# 105 software information middle

# Add/Edit software - Software information - middle section

Closes #105 

Changes proposed in this pull request:
*  Some improvements in the layout and font choice (Roboto is set to first place to closely resemble legacy layout)
*  Middle section of the software information step (see design below).

How to test:

*  `docker-compose down --volumes`: remove older version of database. This PR implements two changes in the database, add new columns to software table and additional view
* `docker-compose up`: it should build database and frontend images (new versions) and create database with new structure 
* Run data-migration again
```bash
# navigate to data-migration folder
cd data-migration
# run data migration
docker-compose up
```
* You are NOT able to add or edit software if you are not signed in. You will see simple 401 message if you try to edit Xenon for example using [edit url](http://localhost/software/xenon/edit)
* Login with SURF using Test IsP and user professor1
* You are NOT able to edit software if you are not maintaining it. For example [Xenon](http://localhost/software/xenon/edit). Currently you will see simple 403 message. Note! Legacy software does not have maintainers assigned to it. When creating new software you are assigned as maintainer (owner) of that software.
* Create new software, using + button at the top right of the screen. After saving you will be redirected to edit page. 
* Provide additional information about software, URL's and markdown. **Check labels on spelling and clarity.** It is possible to use remote markdown file. The content will be loaded in the preview (see image).

![image](https://user-images.githubusercontent.com/9204081/152500526-83aa0ebb-c51c-4e37-a657-edb99d747dcb.png)

* Save changes. You will get notification about success or failure.
* Currently you can only view new software manually, as "published" state is not covered yet. To view software manually use url without "edit" segment, for example: for Xenon **edit url** is `http://localhost/software/xenon/edit` and **view url** is `http://localhost/software/xenon`. 
* Software page is updated to load `remote markdown` files too
* Notification about "unsaved" changes. If you navigate to next section of edit without saving shanges you will get simple notification about it. Further improvements of this feature are needed in order to navigating away from edit section or reloading the page

PR Checklist:
